src/index.js

```diff
@@ -0,0 +1,367 @@
+
+                const btn = document.querySelector('.game-mode');
+                btn.classList.add('active');
+
+                const firstStepBtns = document.querySelectorAll('.first-step button');
+                firstStepBtns.forEach(btn => btn.classList.remove('active'));
+              }
+            }
+          >Start new game
+          </button>
+          <button
+            className={`game-reverse ${!this.state.stepNumber ? 'hide' : ''}`}
+            onClick={() => this.setState({reverseListOfMoves: !this.state.reverseListOfMoves})}
+          >Reverse list
+          </button>
+          <ol className={`game-list ${!this.state.stepNumber ? 'hide' : ''}`}>
+            <li className="game-list-toggle">
+              <button
+                onClick={(e) => {
+                  const list = e.currentTarget.closest('ol');
+
+                  if (list.classList.contains('open')) {
+                    list.classList.remove('open');
+                  } else {
+                    list.classList.add('open');
+                  }
+                }}
+              >
+                List of Moves
+              </button>
+            </li>
+            {moves}
+          </ol>
+        </div>
+      </div>
+    );
+  }
+}
+
+class Board extends React.Component {
+  renderSquare(i, className) {
+    return (
+      <Square
+        className={className || 'square'}
+        value={this.props.squares[i]}
+        onClick={() => this.props.onClick(i)}
+        key={i}
+      />
+    )
+  }
+
+  render() {
+    let squares = [];
+    let rows = [];
+
+    for (let i = 0; i < 9; i++){
+      let className = 'square';
+
+      const winnerLine = this.props.winnerLine;
+      if (winnerLine && winnerLine.includes(i)) {
+        className = 'square square-win';
+      }
+
+      squares.push(this.renderSquare(i, className));
+    }
+
+    for (let i = 0; i < 3; i++){
+      rows.push(squares.slice(i*3, i*3+3))
+    }
+
+    return (
+      <div className="board-container">
+        <div className="board-row">
+          {rows[0]}
+        </div>
+        <div className="board-row">
+          {rows[1]}
+        </div>
+        <div className="board-row">
+          {rows[2]}
+        </div>
+      </div>
+    );
+  }
+}
+
+function GameModeBtm(props){
+  return (
+    <button
+      className={`game-mode ${props.active}`}
+      onClick={() => {
+          props.onClick();
+        }
+      }
+    >Click to play with {props.text}
+    </button>
+  )
+}
+
+function FirstStepRight(props){
+  return (
+    <div className={`first-step ${props.hide} ${props.lock}`}>
+      <h3 className="first-step__title">
+        Who will go first?
+      </h3>
+      <div className="first-step__btns">
+        <button
+          className="first-step__man-btn"
+          onClick={(e) => {
+              props.manFirst();
+              e.currentTarget.classList.add('active');
+              e.currentTarget.nextElementSibling.classList.remove('active');
```

Comment:
it can be done easier, most of the time we don't use such approach in React except very rare cases

Reply:
Thanks, I fixed it:
```
<button
  className={`first-step__man-btn ${props.playerGoesFirst ? 'active' : ''}`}
  onClick={() => {
      props.manFirst();
    }
  }
>
```